From 8705b719573d2159adde10af9c6a4d8806b7d27b Mon Sep 17 00:00:00 2001
From: Leah Rowe <info@minifree.org>
Date: Thu, 25 Jan 2024 14:37:30 +0000
Subject: [PATCH 28/30] dell/e6*30: disable the ME device in devicetree

we neuter anyway. disabling it in devicetree will prevent linux
from ever trying to use it or load a driver for it, and thus
might prevent benign error messages from appearing in dmesg.

this change was suggested by nicholas when asked on irc.

Signed-off-by: Leah Rowe <info@minifree.org>
---
 src/mainboard/dell/e6430/devicetree.cb | 4 ++--
 src/mainboard/dell/e6530/devicetree.cb | 4 ++--
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/mainboard/dell/e6430/devicetree.cb b/src/mainboard/dell/e6430/devicetree.cb
index 054b01c5ac..2b8574c984 100644
--- a/src/mainboard/dell/e6430/devicetree.cb
+++ b/src/mainboard/dell/e6430/devicetree.cb
@@ -39,10 +39,10 @@ chip northbridge/intel/sandybridge # FIXME: GPU registers may not always apply.
 			register "xhci_switchable_ports" = "0x0000000f"
 
 			device ref xhci		on end
-			device ref mei1		on end
+			device ref mei1		off end
 			device ref mei2		off end
 			device ref me_ide_r	off end
-			device ref me_kt	on end
+			device ref me_kt	off end
 			device ref gbe		on end
 			device ref ehci2	on end
 			device ref hda		on end
diff --git a/src/mainboard/dell/e6530/devicetree.cb b/src/mainboard/dell/e6530/devicetree.cb
index 37135bcf0f..010200bb6d 100644
--- a/src/mainboard/dell/e6530/devicetree.cb
+++ b/src/mainboard/dell/e6530/devicetree.cb
@@ -39,10 +39,10 @@ chip northbridge/intel/sandybridge # FIXME: GPU registers may not always apply.
 			register "xhci_switchable_ports" = "0x0000000f"
 
 			device ref xhci		on end # USB 3.0 Controller
-			device ref mei1		on end # Management Engine Interface 1
+			device ref mei1		off end # Management Engine Interface 1
 			device ref mei2		off end # Management Engine Interface 2
 			device ref me_ide_r	off end # Management Engine IDE-R
-			device ref me_kt	on end # Management Engine KT
+			device ref me_kt	off end # Management Engine KT
 			device ref gbe		on end # Intel Gigabit Ethernet
 			device ref ehci2	on end # USB2 EHCI #2
 			device ref hda		on end # High Definition Audio
-- 
2.39.2

